src/Microsoft.CodeAnalysis.Analyzers/Core/MetaAnalyzers/CompareSymbolsCorrectlyAnalyzer.cs

```diff
@@ -38,15 +44,61 @@ public override void Initialize(AnalysisContext context)
             {
                 var compilation = context.Compilation;
                 var symbolType = compilation.GetTypeByMetadataName(s_symbolTypeFullName);
+
                 if (symbolType is null)
                 {
                     return;
                 }
 
                 context.RegisterOperationAction(context => HandleBinaryOperator(in context, symbolType), OperationKind.BinaryOperator);
+
+                var comparerTypes = s_comparerTypeFullName
+                    .Select(comparerTypeFullName => compilation.GetTypeByMetadataName(comparerTypeFullName))
+                    .WhereNotNull()
+                    .ToImmutableArray();
+
+                if (comparerTypes.IsDefaultOrEmpty)
+                {
+                    return;
+                }
+
+                context.RegisterOperationAction(context => HandleOperation(in context, comparerTypes), OperationKind.ObjectCreation);
             });
         }
 
+        private void HandleOperation(in OperationAnalysisContext context, ImmutableArray<INamedTypeSymbol> comparerTypes)
+        {
+            if (context.Operation is IObjectCreationOperation objectCreation)
+            {
+                var typeBeingCreated = (INamedTypeSymbol)objectCreation.Type;
+
+                // Does the type have constructors that take comparers? 
+                var constructorsWithComparers = typeBeingCreated.Constructors.WhereAsArray(constructor => hasSymbolComparer(constructor, comparerTypes));
```

Comment:
😟  I'm concerned about the performance impact of the capturing allocations running on every parameter of every constructor of every object creation...

Reply:
Is there a good way to test the perf impact of this? I'd be happy to run through tests. 

Maybe we could reduce this to types that have generics with `ISymbol` ? That would cover all the cases being tested at the very least. It would miss any specialized types that specifically implement `IDictionar<ISymbol, string>` for example, but it's possible they wouldn't have the correct constructor anyways.